### PR TITLE
Make functions return Result and print debug information on errors

### LINKED ISSUE #108: decide on printing / debugging frequency in the contract
Should we print something for every call ? Or only certain ones?

### PR DESCRIPTION
**Pull request by goastler**
Fixes #108 , #107

### CODE REVIEW

File: contracts/lib.rs [hunk 1]
```diff
@@ -429,11 +462,11 @@ pub mod prosopo {
             let provider_account = self.env().caller();
             // this function is for registration only
             if self.providers.get(&provider_account).is_some() {
-                return Ok(());
+                return err!(Error::ProviderDoesNotExist);
```
Comment (forgetso) [inline contracts/lib.rs:465]: this should be `ProviderExists` or something like that

File: contracts/lib.rs [hunk 2]
```diff
@@ -1080,17 +1105,22 @@ pub mod prosopo {
 
         /// Add an operator
         #[ink(message)]
-        pub fn add_prosopo_operator(&mut self, operator_account: AccountId) {
+        pub fn add_prosopo_operator(&mut self, operator_account: AccountId) -> Result<(), Error> {
```
Comment (forgetso) [inline contracts/lib.rs:1108]: This function should be removed. It must've been added back in by a merge. Please delete it.

**Review by forgetso — CHANGES_REQUESTED** [review]
Really great work. Lots of nice shorthand introduced instead of the more verbose `match`. Only a couple of minor points to address.

File: contracts/lib.rs (hunk 1 above)
Comment (goastler) [reply contracts/lib.rs:500]: oops, my bad! will fix :)

Comment (goastler) [reply contracts/lib.rs]: fixed

File: contracts/lib.rs (hunk 2 above)
Comment (goastler) [reply contracts/lib.rs]: fixed

**Review by forgetso — APPROVED** [review]

**Pull request merged**